crates/subspace-networking/src/connected_peers.rs

```diff
@@ -275,13 +265,18 @@ impl NetworkBehaviour for Behaviour {
             FromSwarm::ConnectionEstablished(ConnectionEstablished { peer_id, .. }) => {
                 // Connection was established without dialing from this protocol
                 if let Entry::Vacant(entry) = self.known_peers.entry(peer_id) {
-                    entry.insert(PeerDecision::PendingDecision {
+                    entry.insert(PeerConnectionState::Deciding {
                         until: Instant::now().add(self.config.decision_timeout),
                     });
 
                     trace!(%peer_id, "Pending peer decision...");
                     self.wake();
                 }
+
+                self.peer_connection_counters
+                    .entry(peer_id)
+                    .and_modify(|counter| *counter += 1)
+                    .or_insert(1);
```

Comment:
Why is this not part of `self.known_peers`? It will access completely different region of memory for no reason while we already have an entry in `self.known_peers` and in many cases when you touch one you need to touch the other too anyway.